cyclonedx_py/parser/poetry.py

```diff
@@ -33,6 +35,7 @@ class PoetryParser(BaseParser):
 
     def __init__(
             self, poetry_lock_contents: str, use_purl_bom_ref: bool = False,
+            omit_category: Set[str] = set(),  # noqa: B006
```

Comment:
what is a category? in poetry they are called groups. 
see https://python-poetry.org/docs/managing-dependencies/#dependency-groups